src/app/CommandSender.cpp

```diff
@@ -337,6 +340,20 @@ void CommandSender::Close()
     mTimedRequest     = false;
     MoveToState(State::AwaitingDestruction);
 
+    if (mpPendingResponseTracker)
```

Comment:
In the cases of timeout waiting for a response, do we want to be sending out `NoCommandResponse`? Because that is what is happening here. If this is not what we want to do, I can change this so we only flush out `NoCommandResponse` to outstanding responses when we get the last `InvokeResponseMessage`.

Reply:
I would have thought timeout waiting for a response would error out the overall interaction, instead of synthesizing per-path bits.